pkg/systemd/hw-detect.js

```diff
@@ -61,6 +63,62 @@ function findPCI(udevdb, info) {
     }
 }
 
+function findMemoryDevices(udevdb, info) {
+    const memoryArray = [];
+    const dmipath = '/devices/virtual/dmi/id';
+    if (!(dmipath in udevdb))
+        return;
+
+    const props = udevdb[dmipath];
+    // Systemd now exposes memory information in udev, introduced in systemd => 248
+    // https://github.com/systemd/systemd/blob/main/NEWS#L1713
+    if (!('MEMORY_ARRAY_NUM_DEVICES' in props)) {
+        return;
+    }
+
+    for (let slot = 0; slot < parseInt(props.MEMORY_ARRAY_NUM_DEVICES, 10); slot++) {
+        let memorySize = parseInt(props[`MEMORY_DEVICE_${slot}_SIZE`], 10);
+        if (memorySize) {
+            memorySize = cockpit.format_bytes(memorySize);
+        } else {
+            memorySize = _("Unknown");
+        }
+
+        let memoryRank = props[`MEMORY_DEVICE_${slot}_RANK`] || _("Unknown");
```

Comment:
DRY: Drop the `||` here, as the `else` now sets this "Unknown" string anyway. Avoids the strings going out of sync, and calling gettext twice.